src/tikzplotly/_save.py

```diff
@@ -64,12 +72,18 @@ def get_tikz_code(
                 data_str.append( "\\addplot coordinates {};\n" )
                 continue
 
-            if trace.x is None:
-                trace.x = list(range(len(trace.y)))
-            if trace.y is None:
-                trace.y = list(range(len(trace.x)))
-
             data_name_macro, y_name = data_container.add_data(trace.x, trace.y, trace.name)
+
+            # If x is textual => symbolic x coords
+            if all(isinstance(v, str) for v in trace.x):
```

Comment:
When `trace.x` is empty, this results in the error `TypeError: 'NoneType' object is not iterable`.
Figures with empty data may occur (I do not now how, but as I must have faced this in the past, as I have implemented [the test](https://github.com/jsaussereau/tikzplotly/blob/17652053cb5e86d2d60e076d6c9ee184eacd5ad1/tests/test_scatter.py#L151))

Reply:
solved in 8700a6af4a8f2208b78addb5f2260fa4f68b2777